config/application.rb

```diff
@@ -20,5 +20,22 @@ class Application < Hanami::Application
       level: :debug,
       stream: settings.log_to_stdout ? $stdout : "log/#{Hanami.env}.log"
     }
+
+    assets_server = settings.assets_server_url
+    config.actions.default_headers['Content-Security-Policy'] = \
+      "base-uri 'self'; " \
+      "child-src 'self'; " \
+      "connect-src 'self'; " \
+      "default-src 'none'; " \
+      "font-src 'self'; " \
+      "form-action 'self'; " \
+      "frame-ancestors 'self'; " \
+      "frame-src 'self'; " \
+      "img-src 'self' https: data:; " \
+      "media-src 'self'; " \
+      "object-src 'none'; " \
+      "plugin-types application/pdf; " \
+      "script-src 'self' #{assets_server}; " \
+      "style-src 'self' 'unsafe-inline' https: #{assets_server}"
```

Comment:
the default value for `settings.assets_server_url` is `http://localhost:8080` so this will not allow `http:`

```suggestion
      "style-src 'self' 'unsafe-inline' #{assets_server.split('//').first} #{assets_server}"
```

```irb
irb(main):001:0> "http://localhost:8080".split('//').first
=> "http:"
irb(main):002:0> "https://localhost:8080".split('//').first
=> "https:"
```

Reply:
Maybe generic `http:`/`https:` should be dropped altogether? I'm no expert on this. It probably does not make sense to have both of them, because more generic `assets_server.split('//').first` will always double `assets_server`.